Code/GraphMol/AddHs.cpp

```diff
@@ -571,6 +572,43 @@ void removeHs(RWMol &mol, bool implicitOnly, bool updateExplicitCount,
     if ((*atIt)->getAtomicNum() == 1) continue;
     (*atIt)->updatePropertyCache(false);
   }
+
```

Comment:
I think it would make more sense to have a function in `SubstanceGroups.h` like `getSubstanceGroupsForAtom()`. This would be generally useful and would help here

Reply:
I thought about using something like that, but then decided for this approach, as it should have a better performance when working on large collections of molecules: SGroups would only be iterated over once per molecule, while `getSubstanceGroupsForAtom()` would iterate over them once per atom (unless cached).

Reply:
Perhaps add `getSubstanceGroupsForAtoms()` as well then?
That could return `std::vector<std::vector<SubstanceGroup *>>` or something similar

Reply:
I would go for that one, maybe with an `inAtomListOnly` (or any better name, that's just what came to mind first) boolean parameter to check only the atom lists (the atom is part of the group), or all SGroup properties (the atom is part of the group, or somehow related to it).

Reply:
makes sense, let's think about this for the next release.